process/smartContract/processorV2/vmInputV2.go

```diff
@@ -35,7 +35,12 @@ func (sc *scProcessor) createVMDeployInput(tx data.TransactionHandler) (*vmcommo
 
 func (sc *scProcessor) initializeVMInputFromTx(vmInput *vmcommon.VMInput, tx data.TransactionHandler) error {
 	var err error
-
+	origScr, ok := (tx).(*smartContractResult.SmartContractResult)
+	if ok {
```

Comment:
make a small separate function - getOriginalCaller - and do the same if else there with return.

Reply:
done